src/inmanta/protocol/endpoints.py

```diff
@@ -293,9 +293,9 @@ async def dispatch_method(self, transport: client.RESTClient, method_call: commo
                 "No such method",
             )
             LOGGER.error(msg)
-            self.add_background_task(
-                self._client.heartbeat_reply(self.sessionid, method_call.reply_id, {"result": msg, "code": 500})
-            )
+            if method_call.reply_id is not None:
+                await self._client.heartbeat_reply(self.sessionid, method_call.reply_id, {"result": msg, "code": 500})
```

Comment:
the background task was a bit moot, as this IS a background task

Reply:
Is it? `Client._call` / `RESTClient.call` wait for the HTTP request to complete, don't they?

Reply:
Also, not specific to this PR but shouldn't this use the `self._heartbeat_client`?

Reply:
1. yes it is, the method calling into this is ran as a background task
2. No, the `_heartbeat_client` is reserved for the forward direction, to make sure it is always available. This reverse direction can generate significant traffic volumes, so it should explicitly kept of that pool.